src/bejeweled/game/GameLogic.java

```diff
@@ -117,23 +118,55 @@ public void returnFromAnimation() {
 	 * untill there are no more combinations then the animations will end
 	 */
 	private void checkForCombinations() {
-		ArrayList<Gem> combinations = board.checkForCombinations();
-		for(Gem g : combinations) {
-			if(g instanceof DeleteRowGem) {
-				combinations = board.deleteRowAndCol(g, combinations);
+		ArrayList<Combination> combinations = board.checkForCombinations();
+		boolean br = false;
+		for (Combination b : combinations) {
+			br = false;
+			for (Gem g : b.getGems()) {
+				if (g instanceof DeleteRowGem) {
+					Combination combi = board.deleteRowAndCol(g, combinations);
+					combinations.add(combi);
+					br = true;
+				}
+			}
+			if (br) {
 				break;
 			}
 		}
-		//combinations found
+		while(br) {
+			br = false;
+			for(Gem g : combinations.get(combinations.size()-1).getGems()) {
+				if (g instanceof DeleteRowGem) {
+					Combination combi = board.deleteRowAndCol(g, combinations);
+					combinations.add(combi);
+					br = true;
+				}
+				if(br) {
```

Comment:
They don't do the same, the while loop only looks at the last element of the ArrayList. I did think alot about this but this seemed the best way to me to implement this.